TypeCobol/Compiler/Nodes/Node.cs

```diff
@@ -44,6 +45,37 @@ public abstract class Node : IVisitable, ICloneable{
         [CanBeNull]
         protected abstract CodeElement InternalCodeElement {  get;}
 
+        /// <summary>
+        /// The Semantic data of this Code Element, usually type information.
+        /// This is a Weak reference because the data can be hold elsewhere, and it 
+        /// can be garbage collected at any moment.
+        /// </summary>
+        System.WeakReference _mySemanticData = null;
```

Comment:
I don't understand. In our current model all instances of every objects are always hold together.
In which case data could be hold elsewhere ?

Reply:
The fact is that the semantic data is build elsewhere and attached here, it is owned by the table symbol, if someone release the symbol table, this reference will be dummy, that's why.